src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -647,35 +647,37 @@ public void SetMembrane(string membraneName)
         gui.UpdateMembraneButtons(Membrane.InternalName);
     }
 
-    public void SetRigidity(float rigidity)
+    public void SetRigidity(int rigidity)
     {
-        if (Math.Abs(Rigidity - rigidity) < MathUtils.EPSILON)
+        int intRigidity = (int)(Rigidity * 10);
+
+        if (Math.Abs(intRigidity - rigidity) < MathUtils.EPSILON)
             return;
 
-        var cost = (int)(Math.Abs(rigidity - Rigidity) / 2 * 100);
+        int cost = (int)(Math.Abs(rigidity - intRigidity) * Constants.MEMBRANE_RIGIDITY_COST_PER_STEP);
 
         if (cost > 0)
         {
```

Comment:
This check can be removed as cost is non-zero always when rigidity != intRigidity